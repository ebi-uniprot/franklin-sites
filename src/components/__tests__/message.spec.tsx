import { render, fireEvent, screen } from '@testing-library/react';

import Message from '../message';

describe('Message component', () => {
  it('should render', () => {
    const { asFragment } = render(
      <Message>
        <div>Some content</div>
      </Message>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with a different level', () => {
    const { asFragment } = render(
      <Message level="failure">
        <div>Some content</div>
      </Message>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should trigger a dimiss callback', () => {
    const dismissFn = jest.fn();
    render(
      <Message onDismiss={dismissFn}>
        <div>Some content</div>
      </Message>
    );
    fireEvent.click(screen.getByRole('button'));
    expect(dismissFn).toHaveBeenCalled();
  });

  it('should render with a subtitle', () => {
    const { asFragment } = render(
      <Message subtitle={<div>test</div>}>
        <div>Some content</div>
      </Message>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render without default icon', () => {
    const { asFragment } = render(
      <Message noIcon>
        <div>Some content</div>
      </Message>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
