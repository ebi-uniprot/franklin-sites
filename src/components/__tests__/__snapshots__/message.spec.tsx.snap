// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component should render 1`] = `
<DocumentFragment>
  <div
    class="message message--info"
    role="status"
  >
    <div
      class="message__side-border"
    />
    <test-file-stub
      classname="message--icon-align-top"
      height="1.125em"
      width="1.125em"
    />
    <div
      class="message__title"
    >
      <div>
        Some content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Message component should render with a different level 1`] = `
<DocumentFragment>
  <div
    class="message message--failure"
    role="status"
  >
    <div
      class="message__side-border"
    />
    <test-file-stub
      classname="message--icon-align-top"
      height="1.125em"
      width="1.125em"
    />
    <div
      class="message__title"
    >
      <div>
        Some content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Message component should render with a subtitle 1`] = `
<DocumentFragment>
  <div
    class="message message--info"
    role="status"
  >
    <div
      class="message__side-border"
    />
    <test-file-stub
      classname="message--icon-align-top"
      height="1.125em"
      width="1.125em"
    />
    <div
      class="message__title"
    >
      <div>
        Some content
      </div>
    </div>
    <div
      class="message__subtitle"
    >
      <div>
        test
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Message component should render with a title passed as prop 1`] = `
<DocumentFragment>
  <div
    class="message message--info"
    role="status"
  >
    <div
      class="message__side-border"
    />
    <test-file-stub
      classname="message--icon-align-center"
      height="1.125em"
      width="1.125em"
    />
    <div
      class="message__title"
    >
      <h1>
        Test
      </h1>
    </div>
    <div
      class="message__text"
    >
      <div>
        Some content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Message component should render without default icon 1`] = `
<DocumentFragment>
  <div
    class="message message--info"
    role="status"
  >
    <div
      class="message__side-border"
    />
    <div
      class="message__title"
    >
      <div>
        Some content
      </div>
    </div>
  </div>
</DocumentFragment>
`;
